File: OrnaTrack/FormView/DgvForExistingPhNo.vb
Public Class DgvForExistingPhno

    Public Property CurrentCustomer As Customer = Nothing
    Public Property CurrentPhNo As String()
    Public Property SelectedCustomer As Integer = -1
    Private Sub DgvForExistingPhno_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dgv.Rows.Clear()

        For Each ID As Integer In Customer.SharedPhNo(CurrentPhNo)
            Dim Record As New Customer(ID)
            If CurrentCustomer IsNot Nothing AndAlso CurrentCustomer.CustomerID = Record.CustomerID Then Continue For

            Dgv.Rows.Add()
            Dgv.Rows(Dgv.Rows.Count - 1).Cells(0).Value = Record.FullName
            Dgv.Rows(Dgv.Rows.Count - 1).Cells(1).Value = Record.PhNo(",")
            Dgv.Rows(Dgv.Rows.Count - 1).Cells(3).Value = Record.CustomerID
        Next
    End Sub

    Private Sub Dgv_CellClick(sender As Object, e As DataGridViewCellEventArgs) Handles Dgv.CellClick
        Try
            If e.RowIndex >= 0 AndAlso e.ColumnIndex = 2 Then
                SelectedCustomer = Dgv.Rows(e.RowIndex).Cells(3).Value
                Close()
            End If
        Catch ex As Exception
            MessageBox.Show("DgvForExistingPhNo/CellClick Error: " & ex.Message)
        End Try
    End Sub
End Class

File: FirstApp/FormView/DgvForExistingPhNo.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

namespace FirstApp
{
    public partial class DgvForExistingPhno
    {

        public Customer CurrentCustomer { get; set; } = null;
        public string[] CurrentPhNo { get; set; }
        public int SelectedCustomer { get; set; } = -1;

        public DgvForExistingPhno()
        {
            InitializeComponent();
            Dgv = _Dgv;
            _Dgv.Name = "Dgv";
        }
        private void DgvForExistingPhno_Load(object sender, EventArgs e)
        {
            Dgv.Rows.Clear();

            foreach (int ID in Customer.SharedPhNo(CurrentPhNo))
            {
                var Record = new Customer(ID);
                if (CurrentCustomer is not null && Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(CurrentCustomer.CustomerID, Record.CustomerID, false)))
                    continue;

                Dgv.Rows.Add();
                Dgv.Rows[Dgv.Rows.Count - 1].Cells[0].Value = Record.FullName();
                Dgv.Rows[Dgv.Rows.Count - 1].Cells[1].Value = Record.PhNo(",");
                Dgv.Rows[Dgv.Rows.Count - 1].Cells[3].Value = Record.CustomerID;
            }
        }

        private void Dgv_CellClick(object sender, DataGridViewCellEventArgs e)
        {
            try
            {
                if (e.RowIndex >= 0 && e.ColumnIndex == 2)
                {
                    SelectedCustomer = Conversions.ToInteger(Dgv.Rows[e.RowIndex].Cells[3].Value);
                    Close();
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show("DgvForExistingPhNo/CellClick Error: " + ex.Message);
            }
        }
    }
}